### PR TITLE
Sync changes from VS2015

### PR DESCRIPTION
**Pull request by AndyGerlicher**
Last of the major changes from initial sync and what we shipped for RTM. Most of this just is a direct copy using Beyond Compare.

### CODE REVIEW

**Comment by AndyGerlicher** [comment]
This closes #171

File: src/XMakeBuildEngine/UnitTests/QaTests/ITestDataProvider.cs [hunk 1]
```diff
@@ -12,14 +15,14 @@
 namespace Microsoft.Build.UnitTests.QA
 {
     /// <summary>
-    /// An interfacing representing a build request definition cache.
+    /// An interfacing representing a build request defination cache.
```
Comment (ValMenn) [inline src/XMakeBuildEngine/UnitTests/QaTests/ITestDataProvider.cs]: Spelling?

File: src/XMakeBuildEngine/UnitTests/QaTests/Integration_Tests.cs [hunk 2]
```diff
@@ -204,7 +207,7 @@ public void OutputFromTaskUpdatesProperty()
 
             r1.WaitForResults();
             ProjectPropertyInstance property = projectInstance.GetProperty("SomeProperty");
-            Assert.IsTrue(property.EvaluatedValue == "Foo", "SomeProperty=Foo");
+            Assert.IsTrue(property.EvaluatedValue == "Foobar", "SomeProperty=FooBar");
```
Comment (ValMenn) [inline src/XMakeBuildEngine/UnitTests/QaTests/Integration_Tests.cs]: Is different case (Foobar vs FooBar) intentional?

File: src/XMakeBuildEngine/UnitTests/QaTests/MockResultsCache.cs [hunk 3]
```diff
@@ -1,26 +1,30 @@
...
 using Microsoft.Build.Evaluation;
 using Microsoft.Build.Collections;
+
 namespace Microsoft.Build.UnitTests.QA
 {
     /// <summary>
     /// Mock Implementation of the results cache which does captures some counters and provides it
     /// back to the tests for validation purposes. Most of the implementation is routed to the default component
-    /// by means of aggregation
+    /// by means of aggrigration
```
Comment (ValMenn) [inline src/XMakeBuildEngine/UnitTests/QaTests/MockResultsCache.cs]: Spelling?

File: src/XMakeBuildEngine/UnitTests/QaTests/RequestDefinition.cs [hunk 4]
```diff
@@ -26,87 +29,87 @@ internal class RequestDefinition : IDisposable
...
 
         /// <summary>
         /// The BuildRequsetEngine
         /// </summary>
-        private IResultsCache resultsCache;
+        private IResultsCache _resultsCache;
 
         /// <summary>
-        /// File name associated to this definition
+        /// File name assiciated to this definition
```
Comment (ValMenn) [inline src/XMakeBuildEngine/UnitTests/QaTests/RequestDefinition.cs]: Spelling

**Comment by ValMenn** [comment]
LGTM.
You might look for some extra comments -- </copyright>, <owner>, and bug references.

**Pull request merged**